Mathlib/Analysis/Analytic/Basic.lean

```diff
@@ -1765,43 +1730,42 @@ to `f x` in a neighborhood of `x`.-/
 theorem HasFiniteFPowerSeriesAt.eventually_const_of_bound_one
     (hf : HasFiniteFPowerSeriesAt f pf x 1) : f =ᶠ[𝓝 x] (fun _ => f x) :=
   Filter.eventuallyEq_iff_exists_mem.mpr (let ⟨r, hf⟩ := hf; ⟨EMetric.ball x r,
-  EMetric.ball_mem_nhds x hf.r_pos, fun y hy ↦ hf.eq_const_of_bound_one y hy⟩)
+    EMetric.ball_mem_nhds x hf.r_pos, fun y hy ↦ hf.eq_const_of_bound_one y hy⟩)
+
+section do_we_really_want_these_lemmas?
 
 /-- If a function admits a finite power series expansion on a disk, then it is continuous there. -/
 protected theorem HasFiniteFPowerSeriesOnBall.continuousOn
     (hf : HasFiniteFPowerSeriesOnBall f p x n r) :
-    ContinuousOn f (EMetric.ball x r) := by
-  apply ContinuousOn.congr ((p.partialSum_continuous n).comp
-    (continuous_id.sub (continuous_const (b := x)))).continuousOn
-  intro y hy
-  simp only [id_eq, Function.comp_apply]
-  exact HasFiniteFPowerSeriesOnBall.eqPartialSum' hf y hy n (le_refl n)
+    ContinuousOn f (EMetric.ball x r) := hf.1.continuousOn
 
 protected theorem HasFiniteFPowerSeriesAt.continuousAt (hf : HasFiniteFPowerSeriesAt f p x n) :
-    ContinuousAt f x :=
-  let ⟨_, hr⟩ := hf
-  hr.continuousOn.continuousAt (EMetric.ball_mem_nhds x hr.r_pos)
+    ContinuousAt f x := hf.toHasFPowerSeriesAt.continuousAt
 
 protected theorem CPolynomialAt.continuousAt (hf : CPolynomialAt 𝕜 f x) : ContinuousAt f x :=
-  let ⟨_, _, hp⟩ := hf
-  hp.continuousAt
+  hf.analyticAt.continuousAt
 
 protected theorem CPolynomialOn.continuousOn {s : Set E} (hf : CPolynomialOn 𝕜 f s) :
     ContinuousOn f s :=
-  fun x hx => (hf x hx).continuousAt.continuousWithinAt
+  hf.analyticOn.continuousOn
+
+end do_we_really_want_these_lemmas?
 
 /-- Continuously polynomial everywhere implies continuous -/
 theorem CPolynomialOn.continuous {f : E → F} (fa : CPolynomialOn 𝕜 f univ) : Continuous f := by
   rw [continuous_iff_continuousOn_univ]; exact fa.continuousOn
 
+protected theorem FormalMultilinearSeries.sum_of_finite (p : FormalMultilinearSeries 𝕜 E F)
+    {n : ℕ} (hn : ∀ m, n ≤ m → p m = 0) (x : E) :
+    p.sum x = p.partialSum n x :=
+  tsum_eq_sum fun m hm ↦ by rw [Finset.mem_range, not_lt] at hm; rw [hn m hm]; rfl
```

Comment:
This lemma isn't actually used, but seems nice to have. I originally wanted to convert the nested tsum in `changeOrigin_eval_of_finite` into nested Finset.sum using this, but that would be more complicated and have less common with the existing `changeOrigin_eval`.